### DISCUSSION TITLE
Low level calls in move

### DISCUSSION THREAD

**Discussion by shivam2320** [aptos-labs/aptos-developer-discussions#229] [Questions] [answered]
Is it possible to do a Low Level Function call from contract A to contract B using module address , function selector and arguments

Like solidity has low level calls (https://3illbaby.hashnode.dev/low-level-calls-in-solidity), is there anything similar for move?

**Comment by MoonShiesty** [comment, marked as answer]
there's no low-level call interface/dynamic dispatch in move right now

unlike solidity, the move-compiler checks that every function call is valid at compile time

**Reply by shivam2320** [reply]
Are there any alternative solutions to this?
Or are there any plans to support this?

**Reply by Alexzander-Stone** [reply]
It's been brought up a few times, but there's no definitive plans, afaik.

Any chance you can inform us what use-cases you'd need this for, @shivam2320? 😌

**Reply by shivam2320** [reply]
Functionalities such as multicall, flashborrow etc cannot function without low-level call interface and dynamic dispatch. Therefore, Move will limit these functionalities.

**Reply by Alexzander-Stone** [reply]
Do you need low-level call interfaces for this? 

We do have dynamic dispatch in the works, and it seems like it should encompass all your needs without the additional low-level calls.

**Reply by shivam2320** [reply]
Yes dynamic dispatch will work
May I know what is the estimated timeline for dynamic dispatch to go live

**Reply by Alexzander-Stone** [reply]
Sorry @shivam2320, I misspoke on the current goals of dynamic dispatch on Aptos.

Today, there is not a defined timeline on supporting general dynamic dispatch. Instead, we identify common scenarios where builders need dynamic dispatch, and create specific methods that support custom logic for them (one example being, [dispatchable fungible assets](https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-73.md)). We believe it's a nice way to build towards dynamic dispatch, and possibly prevent the need for general dynamic dispatch, in the first place.

-------

In regards to the two functionalities that you mentioned... 
Flashloans are already supported/possible on-chain, today. I assume your needs center more around the multi-call process? For arbitrage/swap-related tasks, you'll see developers today create multiple functions on their smart contract to support different lengths and types of swaps. This practice is common across DEX aggregator smart contracts.

##### e.g. function swap paths
function 1: `token1 --DEX1--> token2 --DEX2--> token1`
function 2: `token1 --DEX2--> token2 --DEX1--> token1`
function 3: `token1 --DEX1--> token2 --DEX2--> token3 --DEX2--> token1`
...

But, let us know your thoughts on this! Or if there's any other use cases that you'd rely on dynamic dispatch for. The more interest in a feature from the community, the higher our priority is towards it. 🫡

**Reply by kshitijc1506** [reply]
We have dispatchable fungible asset included in [v1.13.1](https://github.com/aptos-labs/aptos-core/releases/tag/aptos-node-v1.13.1) on testnet now.
